packages/builders/src/components/Assertions.ts

```diff
@@ -8,28 +8,32 @@ export const customIdValidator = s.string
 	.lengthLessThanOrEqual(100)
 	.setValidationEnabled(isValidationEnabled);
 
-export const emojiValidator = s
+const jsonEmojiValidator = s
 	.object({
 		id: s.string,
 		name: s.string,
 		animated: s.boolean,
 	})
 	.partial.strict.setValidationEnabled(isValidationEnabled);
 
+export const emojiValidator = jsonEmojiValidator.nullish;
```

Comment:
```suggestion
export const emojiValidator = s
	.object({
		id: s.string,
		name: s.string,
		animated: s.boolean,
	})
	.partial.strict.nullish.setValidationEnabled(isValidationEnabled);
```

Reply:
I need a separate validator for `jsonOptionValidator`.

Reply:
https://github.com/discordjs/discord.js/pull/8559#discussion_r992410612